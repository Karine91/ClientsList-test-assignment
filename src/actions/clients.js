import $ from "jquery";
import { setDetail } from "./detail";

export const setClients = clients => ({
  type: "SET_CLIENTS",
  clients
});

export const startSetClients = () => {
  return dispatch => {
    return $
      .ajax({
        url: "/clients.json",
        dataType: "json",
        method: "GET"
      })
      .done(data => {
        dispatch(setClients(data));
        dispatch(setDetail(data[0]));
      })
      .fail(err => {
        console.log("error", err);
      });
  };
};
