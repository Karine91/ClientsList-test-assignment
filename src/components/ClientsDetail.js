import React from "react";
import { Item } from "semantic-ui-react";

const ClientsDetail = props => (
  <Item.Group>
    <Item>
      <Item.Image src={props.general && props.general.avatar} />

      <Item.Content>
        <Item.Header>
          {props.general &&
            `${props.general.firstName} ${props.general.lastName}`}
        </Item.Header>
        <Item.Meta>
          {props.job && `${props.job.title} - ${props.job.company}`}
        </Item.Meta>
        <Item.Description>
          {props.address &&
            `${props.address.zipCode}, ${props.address.country} ${
              props.address.city
            } ${props.address.street}`}
        </Item.Description>
        <Item.Extra>
          {props.contact && `${props.contact.email}, ${props.contact.phone}`}
        </Item.Extra>
      </Item.Content>
    </Item>
  </Item.Group>
);

export default ClientsDetail;
