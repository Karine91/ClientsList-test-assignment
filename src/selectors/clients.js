// Get visible expenses
const getValueFromObject = (outputArr, obj) => {
  for (let key in obj) {
    let elem = obj[key];
    if (typeof elem !== "object") {
      if (key !== "avatar") {
        outputArr.push(elem);
      }
    } else {
      getValueFromObject(outputArr, elem);
    }
  }
};
const objectToArray = obj => {
  const textValuesArr = [];
  getValueFromObject(textValuesArr, obj);
  return textValuesArr;
};

export default (clients, { text }) => {
  if (text) {
    return clients.filter(client => {
      const textValuesArr = objectToArray(client);
      let textMatch;
      for (let i = 0; i < textValuesArr.length; i++) {
        let re = new RegExp(text, "i");
        textMatch = textValuesArr[i].match(re);
        if (textMatch) break;
      }
      return textMatch;
    });
  }
  return clients;
};
