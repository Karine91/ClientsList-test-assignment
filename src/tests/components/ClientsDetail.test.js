import React from "react";
import { shallow } from "enzyme";
import ClientsDetail from "../../components/ClientsDetail";
import clients from "../fixtures/clients";

test("should render ClientsDetail", () => {
  const wrapper = shallow(<ClientsDetail {...clients[0]} />);
  expect(wrapper).toMatchSnapshot();
});
