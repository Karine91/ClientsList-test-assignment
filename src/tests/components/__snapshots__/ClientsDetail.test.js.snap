// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ClientsDetail 1`] = `
<ItemGroup>
  <Item>
    <ItemImage
      src="https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg"
    />
    <ItemContent>
      <ItemHeader>
        Liana Crooks
      </ItemHeader>
      <ItemMeta>
        Investor Functionality Coordinator - Ledner, Johnson and Predovic
      </ItemMeta>
      <ItemDescription>
        42586-7898, Guinea-Bissau New Devon 1520 Zemlak Cove
      </ItemDescription>
      <ItemExtra>
        Gerry_Hackett77@gmail.com, (895) 984-0132
      </ItemExtra>
    </ItemContent>
  </Item>
</ItemGroup>
`;
